### DISCUSSION TITLE
线性复数模式下刷字符使用限制

### DISCUSSION THREAD

**Discussion by bczhc** [Physics365/991CN-X-CW-Decompilation#43] [General]
[附：线性复数an模式中刷字符的方法](https://github.com/Physics365/991CN-X-CW-Decompilation/blob/a0c4013/ROP/%E6%95%99%E7%A8%8B/991CN%20X%20(VerC)%20%E5%BC%82%E5%B8%B8%E6%95%99%E7%A8%8B/%E7%AC%AC%E4%B8%89%E7%AB%A0%20%E5%8F%98%E9%87%8F%E4%B8%8E%E5%8E%86%E5%8F%B2%E8%AE%B0%E5%BD%95.md#%E9%99%84%E7%BA%BF%E6%80%A7%E5%A4%8D%E6%95%B0an%E6%A8%A1%E5%BC%8F%E4%B8%AD%E5%88%B7%E5%AD%97%E7%AC%A6%E7%9A%84%E6%96%B9%E6%B3%95)

模拟器操作：

1. 进入线性复数模式
2. 输入
   `422345a5312e30303030333133313331
   333133312d33312340a5312e33323332
   333233323332333233322d33322347a5
   312e3333333333333333333333333333
   2d33332348a5312e3334333433343334
   3334333433342d3334a6312e33353335
   333533353335333533352d333520`
3. 全部执行
4. 载入\<offset\>an
5. 往历史记录里放一条lbf转换器
6. 输入
   `31313123606040d0d02348234da545234fa547`
7. 全部执行后按上键

测试出：
貌似只有当an的偏移大于一定值时这个才能用？上方的操作在110an和124an下都不可用，136an可用。

如果赋值时把x的虚部去掉，即少刷八个字符，那么124an下可用，但110an下不可用。

**Comment by bczhc** [comment]
并且好像依然需要第六字节不能为字母

**Comment by fish4terrisa-MSDSM** [comment]
其实有
```
1.xx xx xx xx xx xx xx x10 xx -> x

x in>cm:
x:
x in>cm:
@(748)=1.0000 xx xx xx xx xx x10 xx
CALCx4
```
的方法的说

**Reply by fish4terrisa-MSDSM** [reply]
（一般够用），是不是可以整理一下这些放到教程的附录里?

**Reply by bczhc** [reply]
哦这个我知道的，，不过六个字节显然不够用😂

**Reply by bczhc** [reply]
不过确实方便太多，在q的知乎里看到的。这个在110an下对x赋值会卡，要124an。

**Reply by fish4terrisa-MSDSM** [reply]
其实不太一样，q的知乎里是
```
F.xx xx xx xx xx xx xx x10 xx ->x
```
但是这个会导致按开机键之后计算器重置

**Reply by Xyzstk** [reply]
如果整数部分不是十六进制数码的话更容易触发进位

**Reply by Xyzstk** [reply]
124an应该最多可以刷30个
```
123:
12345:
D:
@(0x4D)=A:
@(0x4E)=B:
@(0x4F)=C
```

**Reply by fish4terrisa-MSDSM** [reply]
不过进位也没办法的说，毕竟每次开机清空的话要重打rop太伤了

**Reply by Xyzstk** [reply]
屏幕缓冲区之后的内容应该无论如何都不会清空的吧（除非抠电池）

**Comment by bczhc** [comment]
不知道是我做错了还是什么，反正用这个DMFx的得要an大一些才能刷出，110和124的都不管用。倒是在110an下我发现，如果把x的虚部去掉，
```
D=1.0000aaaaaaaaaa[x10]aa
M=1.bbbbbbbbbbbbbb[x10]bb
F=1.cccccccccccccc[x10]cc
x=1.00000000dddddd[x10]dd
```
并且之后用
`111:M:x:@(4D)=D:@(4F)=F`（不给M加括号），可以刷aa bb cc dd共6+8+8+4个

**Comment by EwTE-H** [comment]
看历史记录教程，里面给出了详细原理，跟着推导就行了
